src/dataflow/components/dataflow-program.tsx

```diff
@@ -436,6 +457,31 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     }
   }
 
+  private updateGraphDataSet = (programId: string) => {
+    if (this.state.isProgramRunning) {
+      fetchProgramData(programId).then((result: any) => {
```

Comment:
So at the moment, we're always fetching all the data, is that right? And eventually we'll want to just select the data from certain times, right?

If that's right, it doesn't need to be PR, although it also seems like it wouldn't be too hard. I don't think you even need to store any more information, since you can grab the time of the last data point from the existing datasets, and use that as the start time.

Reply:
Yeah, I'm just grabbing everything at present and doing it on the heartbeat.  I think eventually we will want to differentiate between a completed and a live dataset and only request new data from a live dataset (one that is actively being recorded to).